### ISSUE TITLE
Only USB works no NAT functionality?

### ISSUE THREAD

**Issue by spacesuitdiver** [iamromulan/cellular-modem-wiki#15]
I've tried pretty much everything I can find and I cannot get the modem to connect NAT working for the life of me 🙁

This is the device I am using...
https://www.aliexpress.us/item/3256806989722627.html?spm=a2g0o.order_list.order_list_main.36.2d2a1802DBJvyS&gatewayAdapt=glo2usa

Currently I have everything configured as listed here in this thread.
https://forums.quectel.com/t/rm520n-gl-pcie-to-ethernet-and-dhcp/25029

And another thread I have going when I was having issues with the driver and didn't quite understand the "usbnet" modes properly.
https://forums.quectel.com/t/only-seeing-dm-nmea-ports/37233/1

Thoughts on how to troubleshoot? I ran
```
adb shell ping 8.8.8.8
Network is unreachable
```
and
```
adb ifconfig
[lists bridge0 device w/ 192.168.225.1]
[list rmnet_ipa0 w/ no valid ip]
```

**Comment by iamromulan** [comment]
Set your USB mode to 1
ECM is a AP mode so auto connect will never be an issue. What does the following respond with
AT+QMAPWAC?
AT+QMAP="WWAN" 
AT+QENG="servingcell"
AT+QUIMSLOT?
AT+QSIMSTAT?

Your aliexpress link doesn't seem to work?

**Comment by kubepy** [comment]
Hi, @spacesuitdiver 

I bought the same device as you, I was using usbnet mode 3(RNDIS) firstly, sometime the ethernet was lost and dhcp disappeared, however it looks like usbnet mode to 1 (ECM) is better for my 2nd board test today.

I find this device requires a stable power adapter for working, I used my Raspberry PI 4B power adapter 5.1V * 3A seems it is okay.

When I was using PD adapter, it was not stable and sometime ethernet dhcp is losing and I cannot connect to the devices anymore (finally after many times retry then my 1st board is broken, had sent to after service) I asked the customer support, they answered the PD adapter is NOT support, normal adapter is good.

**Comment by shadowkora** [comment]
Hey @iamromulan, hijacking this thread.

I am running into the same issue of `adb shell ping 8.8.8.8` only showing `Network is unreachable`

Here is an output of a few of my commands:
Model Number : RM500Q-GL
AT+QGMR
RM500QGLABR13A01M4G_01.001.01.001

AT+QMAPWAC?
+QMAPWAC: 1

AT+QMAP="WWAN"
+QMAP: "WWAN",0,1,"IPV4","0.0.0.0"
+QMAP: "WWAN",0,1,"IPV6","0:0:0:0:0:0:0:0"

AT+QENG="servingcell"
+QENG: "servingcell","NOCONN"
+QENG: "LTE","FDD",230,02,DFDA5CA,345,100,1,5,5,722,-96,-10,-67,17,15,230,-
+QENG: "NR5G-NSA",230,02,64,-108,15,-13,644640,78,6,1

AT+QUIMSLOT?
+QUIMSLOT: 1

AT+QSIMSTAT?
+QSIMSTAT: 0,1

Funnily enough, the cellular connection on my laptop works just fine while the ping using adb doesn't, usbnet is set to 2, but I also tried with usbnet set to one and it didn't work either.
Tried it with 2 different boards: https://www.aliexpress.com/item/1005005714195628.html and https://www.aliexpress.com/item/1005003952829784.html (type C adapter)

**Comment by iamromulan** [comment]
I unfortunately don't have a RM500 to test with but...

### USB modes
- When presented as a USB  modem to a device/host there are 4 different USB modes (protocols/standards) that it can present itself as. 2 of them, the modem is presented as a full cellular modem to the host and the host will be in charge (TE). Those modes are NDIS (Referred to as QMI in the past) and MBIM. The other 2; ECM and RNDIS; the modem is in charge (AP) and the host will view it as a USB ethernet port. It will simply ask for an IP address and that's it.
- You can view what USB mode the modem is in with the AT command: ``AT+QCFG="usbnet"``
  - To set it to a particular USB mode:
    - ``AT+QCFG="usbnet",0`` (NDIS/QMI) (TE) (Default)
    - ``AT+QCFG="usbnet",1`` (ECM) (AP)
    - ``AT+QCFG="usbnet",2`` (MBIM) (TE)
    - ``AT+QCFG="usbnet",3`` (RNDIS) (AP)
- :bulb: After setting a USB mode reboot to have it  take effect with ``AT+CFUN=1,1``

The QMAPWAC command is supposed to control automatic connection. This is forced to 1 no matter what in AP modes.

If you can access it on your PC ok in a TE mode but not on the modem itself then QMAPWAC=1 might not be taking effect. After a reboot does QMAPWAC stay 1?

**Comment by iamromulan** [comment]
Oh also ECM mode can be used on Windows. When it reboots after setting usbnet 1 you'll see an ECM Device in device manager with no driver. Install the ECM driver then right click on it in device manager ---> update driver ---> manually choose from a list --->  network interface ---> Quectel  ---> ECM adapter

That'll give you a virtual USB Ethernet adapter provided from the modem.

**Comment by iamromulan** [comment]
The Application Processor (Linux OS on the modem) must be able to reach GitHub servers on the internet. That's the main requirement until I make an offline installer.

In ADB shell do you see a rmnet_data interface appear when you run ifconfig? That's the interface that gets created when a data connection is established.

Could try setting the mpdn rule with IP passthrough off.
AT+QMAP="mpdn_rule",0,1,0,0,1

Basically, Rule 0, VLAN0, IPPT off, auto connect on

**Comment by shadowkora** [comment]
@iamromulan 
Hey, sorry for disappearing, life got busy.

I tried setting the MPDN rule, it didn't work. The interfaces I see are the following
> bridge0
> ecm0
> eth0
> lo
> rmnet_ipa0

Now I went to change `AT+QETH: "RGMII","DISABLE",1,0` to `AT+QETH="RGMII","ENABLE",1,0` as per this post: https://wirelessjoint.com/viewtopic.php?p=25464#p25464 and it fixed the issue, bonus is, that it works with MBIM now as well, no need for ECM.

Successfully installed simpleadmin as well.

Hope this find helps people in similar situation. cc: @spacesuitdiver

**Comment by shadowkora** [comment]
Now the only issue is, that it doesn't connect to network if I only use the RJ45 for some reason

**Comment by shadowkora** [comment]
Aha! Seems the fix/script you have here: https://github.com/iamromulan/quectel-rgmii-toolkit/blob/0b1f3b22d815a2c4a773cca185e6b740629bd6e0/old/original_main_repo/RMxxx_rgmii_toolkit.sh#L817 is not present on my specific device

**Comment by iamromulan** [comment]
Closing
